src/fmt.rs

```diff
@@ -178,21 +195,25 @@ pub unsafe trait CharFormat<'a>: Format {
         F: FnOnce(&[u8]);
 }
 
-/// Indicates a Rust slice type that is represented in memory as bytes.
+/// Indicates a Rust slice type represented in memory as bytes.
+/// 
+/// # Safety
+/// 
+/// It manipulates raw data and may transmute a non-UTF [`&[u8]`] into [`str`]
```

Comment:
This should be of the form "types that implement this MUST ...." but I'm not actually sure what the impl safety guarantees are here

Reply:
I too, am unsure of the safety guarantees here, it's my minimal effort guess.